src/backend/embedding/chunkfile_Weaviateprocess/chunkfile_uploadWeaviate.py

```diff
@@ -0,0 +1,193 @@
+                logger.info(f"Schema {class_name} already exists. Deleting it first...")
+                self.client.schema.delete_class(class_name)
+
+            schema = {
+                "class": class_name,
+                "description": "Schema for storing semantic chunks of a document.",
+                "properties": [
+                    {
+                        "name": "text",
+                        "dataType": ["text"],
+                        "description": "The chunked text content from the document.",
+                        "indexInverted": True
+                    },
+                    {
+                        "name": "chunk_number",
+                        "dataType": ["int"],
+                        "description": "An object containing metadata like chunk number",
+                        "indexInverted": False
+                    },
+                    {
+                        "name": "page_numbers",
+                        "dataType": ["int[]"],
+                        "description": "An object containing metadata like page number",
+                        "indexInverted": False
+                    },
+                    {
+                        "name": "customer_guid",
+                        "dataType": ["text"],
+                        "description": "A unique identifier for the customer (namespace-like isolation).",
+                        "indexInverted": True
+                    },
+                    {
+                        "name": "filename",
+                        "dataType": ["text"],
+                        "description": "The name of the file this chunk originates from.",
+                        "indexInverted": False
+                    }
+                ],
+                "vectorizer": "text2vec-transformers"
+            }
+            self.client.schema.create_class(schema)
+            logger.info(f"Schema {class_name} created successfully!")
+        except Exception as e:
+            logger.error(f"Unexpected error in schema creation: {e}")
+
+class DataManager:
+    def __init__(self, client):
+        self.client = client
+
+    def get_class_name(self):
+        return "SemanticChunkSearch"
+
+    def insert_data_from_file(self, customer_guid: str, file_path: str):
+        try:
+            class_name = self.get_class_name()
+            with open(file_path, "r", encoding="utf-8") as file:
+                data = json.load(file)
+            if not isinstance(data, list):
+                logger.error("Invalid JSON format: Expected a list of objects.")
+            # Insert each data chunk
+            for index, entry in enumerate(data):
+                text = entry.get("text", "").strip()
+                metadata = entry.get("metadata", {})
+
+                chunk_data = {
+                    "text": text,
+                    "customer_guid": customer_guid,
+                    "filename": metadata.get("filename", ""),
```

Comment:
We should not provide optional  fields here
You should throw an exception if the respective field is not present